From 46415802601d23d417736c9bc60fdf97e7a1b034 Mon Sep 17 00:00:00 2001
From: mingdaxu <mingda.xu@intel.com>
Date: Tue, 8 Jan 2019 15:24:45 +0800
Subject: [PATCH 2321/2367] media: intel-ipu4: Fixing static code check issue

Fixing coding issue of IPU driver common code reported by static
check tool

Change-Id: I46120312bb76d4f3e36d4d5780e2140eeaf81451
Tracked-On: PKT-2212
Signed-off-by: mingdaxu <mingda.xu@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
---
 drivers/media/pci/intel/ipu-dma.c | 6 ++++--
 drivers/media/pci/intel/ipu.c     | 5 ++++-
 2 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/drivers/media/pci/intel/ipu-dma.c b/drivers/media/pci/intel/ipu-dma.c
index 0770d8d..8a12c03 100644
--- a/drivers/media/pci/intel/ipu-dma.c
+++ b/drivers/media/pci/intel/ipu-dma.c
@@ -258,7 +258,8 @@ static void ipu_dma_free(struct device *dev, size_t size, void *vaddr,
 	if (WARN_ON(!area->pages))
 		return;
 
-	WARN_ON(!iova);
+	if (WARN_ON(!iova))
+		return;
 
 	size = PAGE_ALIGN(size);
 
@@ -323,7 +324,8 @@ static void ipu_dma_unmap_sg(struct device *dev,
 	if (!nents)
 		return;
 
-	WARN_ON(!iova);
+	if (WARN_ON(!iova))
+		return;
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(4, 8, 0)
 	if (!dma_get_attr(DMA_ATTR_SKIP_CPU_SYNC, attrs))
diff --git a/drivers/media/pci/intel/ipu.c b/drivers/media/pci/intel/ipu.c
index 322ddfb..27d97c9 100644
--- a/drivers/media/pci/intel/ipu.c
+++ b/drivers/media/pci/intel/ipu.c
@@ -330,12 +330,15 @@ int request_cpd_fw(const struct firmware **firmware_p, const char *name,
 		*firmware_p = fw;
 	} else {
 		tmp = kzalloc(sizeof(struct firmware), GFP_KERNEL);
-		if (!tmp)
+		if (!tmp) {
+			release_firmware(fw);
 			return -ENOMEM;
+		}
 		tmp->size = fw->size;
 		tmp->data = vmalloc(fw->size);
 		if (!tmp->data) {
 			kfree(tmp);
+			release_firmware(fw);
 			return -ENOMEM;
 		}
 		memcpy((void *)tmp->data, fw->data, fw->size);
-- 
2.7.4

